src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -403,7 +408,7 @@ protected override void Dispose(bool disposing)
 
                 public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
                 {
-                    return _http2Stream.SendDataAsync(buffer);
+                    return _http2Stream?.SendDataAsync(buffer) ?? default(ValueTask);
```

Comment:
Shouldn't it be an exception if _http2Steam is null?

Reply:
You tell me :)  I was trying to be consistent with the behavior today where it doesn't check disposed at all.  That said, would that case ever occur?  Based on current usage, this stream is never handed out.  Maybe it should just assert?

Reply:
It's passed to the HttpContent object to serialize to. A user can provide their own derived HttpContent class. So it is handed out.

The current behavior isn't correct. We should be checking for disposed.

Reply:
Ok.  I'll throw an ObjectDisposedException if it's been disposed.